docs/content/SegmentedControl.mdx

```diff
@@ -0,0 +1,190 @@
+---
+title: SegmentedControl
+status: Draft
+description: Used to select an option from a short list and immediately apply the selection
+---
+
+## Examples
+
+### Simple
+
+```jsx live drafts
+<SegmentedControl aria-label="File view">
+  <SegmentedControl.Item>Preview</SegmentedControl.Item>
+  <SegmentedControl.Item>Raw</SegmentedControl.Item>
+  <SegmentedControl.Item>Blame</SegmentedControl.Item>
+</SegmentedControl>
+```
+
+### With icons and labels
+
+```jsx live drafts
+<SegmentedControl aria-label="File view">
+  <SegmentedControl.Item leadingIcon={EyeIcon}>Preview</SegmentedControl.Item>
+  <SegmentedControl.Item leadingIcon={FileCodeIcon}>Raw</SegmentedControl.Item>
+  <SegmentedControl.Item leadingIcon={PeopleIcon}>Blame</SegmentedControl.Item>
+</SegmentedControl>
+```
+
+### With icons only
+
+```jsx live drafts
+<SegmentedControl aria-label="File view">
+  <SegmentedControl.IconItem icon={EyeIcon} aria-label="Preview" />
+  <SegmentedControl.IconItem icon={FileCodeIcon} aria-label="Raw" />
+  <SegmentedControl.IconItem icon={PeopleIcon} aria-label="Blame" />
+</SegmentedControl>
+```
+
+### With labels hidden on smaller viewports
+
+```jsx live drafts
+<SegmentedControl aria-label="File view" hideLabels={{narrow: false, large: true}}>
+  <SegmentedControl.Item leadingIcon={EyeIcon}>Preview</SegmentedControl.Item>
+  <SegmentedControl.Item leadingIcon={FileCodeIcon}>Raw</SegmentedControl.Item>
+  <SegmentedControl.Item leadingIcon={PeopleIcon}>Blame</SegmentedControl.Item>
+</SegmentedControl>
+```
+
+### Convert to a dropdown on smaller viewports
+
+```jsx live drafts
+<SegmentedControl aria-label="File view" asDropdown={{narrow: true, large: false}}>
+  <SegmentedControl.Item leadingIcon={EyeIcon}>Preview</SegmentedControl.Item>
+  <SegmentedControl.Item leadingIcon={FileCodeIcon}>Raw</SegmentedControl.Item>
+  <SegmentedControl.Item leadingIcon={PeopleIcon}>Blame</SegmentedControl.Item>
+</SegmentedControl>
+```
```

Comment:
Alternate idea:
```jsx
<SegmentedControl aria-label="File view" variant={{narrow: 'dropdown', regular: 'none'}}>
  <SegmentedControl.Item leadingIcon={EyeIcon}>Preview</SegmentedControl.Item>
  <SegmentedControl.Item leadingIcon={FileCodeIcon}>Raw</SegmentedControl.Item>
  <SegmentedControl.Item leadingIcon={PeopleIcon}>Blame</SegmentedControl.Item>
</SegmentedControl>
```

Reply:
I like the idea this alternate of using variant for dropdown more!

Reply:
What would happen if the variant was "compact" and the buttons didn't have any icons?

Reply:
I don't think anything would happen because there would be no icons to hide